src/index.tsx

```diff
@@ -38,6 +46,13 @@ const DEFAULT_INITIAL_VIEW_STATE = {
 const DEFAULT_MAP_STYLE =
   "https://basemaps.cartocdn.com/gl/positron-nolabels-gl-style/style.json";
 
+function DeckGLOverlay(props: MapboxOverlayProps) {
+  const overlay = useControl(() => new DeckOverlay(props));
+
+  overlay.setProps(props);
+  return null;
+}
```

Comment:
Is this the standard way to use `DeckOverlay` in react? @felixpalmer

Reply:
Yes https://deck.gl/docs/api-reference/mapbox/mapbox-overlay